daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/PState.scala

```diff
@@ -188,7 +189,14 @@ final class PState private (
   def mark(requestorID: String): PState.Mark = {
     // threadCheck()
     val m = markPool.getFromPool(requestorID)
-    m.captureFrom(this, requestorID)
+    try {
+      m.captureFrom(this, requestorID)
+    } catch {
+      case b: BacktrackingException => {
+        markPool.returnToPool(m)
+      }
+      throw b
```

Comment:
I assume this comes from a lock throwing the exception? This would go away too, I think?